pkg/build/controller/image_change_controller_test.go

```diff
@@ -1,427 +0,0 @@
-package controller
-
-import (
-	"errors"
-	"fmt"
-	"strings"
-	"testing"
-
-	kerrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	apirequest "k8s.io/apiserver/pkg/endpoints/request"
-	kapi "k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/fake"
-
-	buildapi "github.com/openshift/origin/pkg/build/api"
-	buildtest "github.com/openshift/origin/pkg/build/controller/test"
-	buildgenerator "github.com/openshift/origin/pkg/build/generator"
-	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
-	imageapi "github.com/openshift/origin/pkg/image/api"
-)
-
-func TestNewImageID(t *testing.T) {
```

Comment:
OK, I just pushed an update that add the new image id and new image id + default tag
as well as an equivalent for the different tag update 2 test.

Given the minor implementation differences noted in https://github.com/openshift/origin/pull/14226#issuecomment-302778566 , the conflict error and fatal error tests do not apply, as they are not special cased in the new, "generic" implementation.

Also, I believe the "build config update" error tests are/will be handled by the "instantiate" test.

The only thing that remains is updating the build instantiator in the new package to have a mock build generator and validate that path.

However I decided to push this update as an intermediate update of sorts.